src/base/tristatebool.h

```diff
@@ -47,7 +49,7 @@ class TriStateBool
     bool operator!=(const TriStateBool &other) const;
 
 private:
-    int m_value = -1; // Undefined by default
+    int8_t m_value = -1; // Undefined by default
```

Comment:
signed char would not require any include.

Reply:
And, if you want to refactor this even more, an enum would fit better. Or replace with boost::logic::tribool.

Reply:
>signed char would not require any include.

`char` was used at first, but I thought `char` is for characters... will change if others agree. 👍 

>And, if you want to refactor this even more, an enum would fit better.

Not looking for this at this moment.

>Or replace with boost::logic::tribool.

No, their equality comparison is not the same as ours.